components/director/internal/domain/labeldef/converter.go

```diff
@@ -15,19 +15,37 @@ func NewConverter() *converter {
 
 type converter struct{}
 
-func (c *converter) FromGraphQL(input graphql.LabelDefinitionInput, tenant string) model.LabelDefinition {
+func (c *converter) FromGraphQL(input graphql.LabelDefinitionInput, tenant string) (model.LabelDefinition, error) {
+	var schema interface{}
+	if input.Schema != nil {
+		err := json.Unmarshal([]byte(*input.Schema), &schema)
+		if err != nil {
+			return model.LabelDefinition{}, err
+		}
+	}
+
 	return model.LabelDefinition{
 		Key:    input.Key,
-		Schema: input.Schema,
+		Schema: &schema,
 		Tenant: tenant,
-	}
+	}, nil
 }
 
-func (c *converter) ToGraphQL(in model.LabelDefinition) graphql.LabelDefinition {
+func (c *converter) ToGraphQL(in model.LabelDefinition) (graphql.LabelDefinition, error) {
+	var jsonSchema graphql.JSON
+	if in.Schema != nil {
```

Comment:
Please invert this condition to avoid deep nesting

Reply:
Maybe should I extract to private method, because if I invert the condition I had return object in `if input.Schema==nil` condition and copy paste (change to `Schema: nil`) the return statement.

Reply:
yup, private method would be a good idea